### PR TITLE
Add Match, Contains, StartsWith and EndsWith filters

### PR DESCRIPTION
**Pull request by N-lson**
Adds four new filters for string validation

### CODE REVIEW

File: source/Octostache/Templates/Functions/TextValidationFunction.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Octostache.Templates.Functions
+{
+    internal static class TextValidationFunction
```
Comment (droyad) [inline source/Octostache/Templates/Functions/TextValidationFunction.cs:5]: Would `TextComparisonFunctions` be better?

File: source/Octostache/Templates/Functions/TextValidationFunction.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+            return Regex.Match(argument, options[0]).Success.ToString().ToLowerInvariant();
+        }
+        
+        public static string? StartsWith(string? argument, string[] options)
+        {
+            if (argument == null || options.Length != 1)
+                return null;
+
+            return argument.StartsWith(options[0]).ToString().ToLowerInvariant();
```
Comment (droyad) [inline source/Octostache/Templates/Functions/TextValidationFunction.cs:20]:
```suggestion
            return argument.StartsWith(options[0], StringComparison.Ordinal).ToString().ToLowerInvariant();
```

File: source/Octostache/Templates/Functions/TextValidationFunction.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+        
+        public static string? EndsWith(string? argument, string[] options)
+        {
+            if (argument == null || options.Length != 1)
+                return null;
+
+            return argument.EndsWith(options[0]).ToString().ToLowerInvariant();
```
Comment (droyad) [inline source/Octostache/Templates/Functions/TextValidationFunction.cs:28]:
```suggestion
            return argument.EndsWith(options[0], StringComparison.Ordinal).ToString().ToLowerInvariant();
```

File: source/Octostache/Templates/Functions/TextValidationFunction.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        }
+        
+        public static string? Contains(string? argument, string[] options)
+        {
+            if (argument == null || options.Length != 1)
+                return null;
+
+            return argument.Contains(options[0]).ToString().ToLowerInvariant();
```
Comment (droyad) [inline source/Octostache/Templates/Functions/TextValidationFunction.cs:36]: This defaults to `Ordinal`. The others likely do, but lets be explicit. There is no StringComparison overload for `Contains` on .NET 4.0

**Review by droyad — APPROVED** [review]

File: source/Octostache/Templates/Functions/TextValidationFunction.cs (hunk 1 above)
Comment (N-lson) [reply source/Octostache/Templates/Functions/TextValidationFunction.cs:5]: Was considering that after I made the name. I agree it makes more sense than validation

File: source/Octostache/Templates/Functions/TextValidationFunction.cs (hunk 4 above)
Comment (N-lson) [reply source/Octostache/Templates/Functions/TextValidationFunction.cs:36]: Thanks, wasn't aware of the overload. Checked the doc for StartsWith and looks like it doesn't use Ordinal so we need to specify it.
image

**Pull request merged**